flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphHasherV2.java

```diff
@@ -285,8 +284,8 @@ private boolean isChainable(StreamEdge edge, boolean isChainingEnabled, StreamGr
 		StreamNode upStreamVertex = streamGraph.getSourceVertex(edge);
 		StreamNode downStreamVertex = streamGraph.getTargetVertex(edge);
 
-		StreamOperator<?> headOperator = upStreamVertex.getOperator();
-		StreamOperator<?> outOperator = downStreamVertex.getOperator();
+		StreamOperatorFactory<?> headOperator = upStreamVertex.getOperatorFactory();
```

Comment:
Is this compatible with previous savepoints? Because we now hash the factory not the operator.

Reply:
The hash value should not be changed:
The hash is either computed from the transformation's user-specified id or generated in a deterministic way. user-specified id not be changed.
The generated hash is deterministic with respect to:
1.node-local properties (node ID): node ID is generated by id counter.
2.chained output nodes: Chaining strategy not be changed.
3.input nodes hashes

So I think not related to StreamOperatorFactory.